src/dataset/mod.rs

```diff
@@ -96,3 +108,144 @@ where
 {
     fn labels(&self) -> Vec<Self::Elem>;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ndarray::{array, Array1, Array2};
+    use ndarray_rand::rand::SeedableRng;
+    use rand_isaac::Isaac64Rng;
+
+    #[test]
+    fn dataset_implements_required_methods() {
+        let mut rng = Isaac64Rng::seed_from_u64(42);
+
+        // ------ Targets ------
+
+        // New
+        let mut dataset = Dataset::new(array![[1., 2.], [1., 2.]], array![0., 1.]);
+
+        // Shuffle
+        dataset = dataset.shuffle(&mut rng);
+
+        // Bootstrap
+        let mut iter = dataset.bootstrap(3, &mut rng);
+        for _ in 1..5 {
+            let b_dataset = iter.next().unwrap();
+            assert_eq!(b_dataset.records().dim().0, 3);
+        }
+
+        let linspace: Array1<f64> = Array1::linspace(0.0, 0.8, 100);
+        let records = Array2::from_shape_vec((50, 2), linspace.to_vec()).unwrap();
+        let targets: Array1<f64> = Array1::linspace(0.0, 0.8, 50);
+        let dataset = Dataset::new(records, targets);
+
+        //Split with ratio view
+        let (train, val) = dataset.split_with_ratio_view(0.5);
+        assert_eq!(train.targets().len(), 25);
+        assert_eq!(val.targets().len(), 25);
```

Comment:
`split_with_ratio_view` for type `Dataset` uses the already existing implementation of the method, but this returns a slice as the targets. Would it be better to re-implement it for `Dataset` so that it returns an `ArrayView1` as the targets?